### PR TITLE
No sound ext

### PR DESCRIPTION
**Pull request by DolceTriade**

### CODE REVIEW

File: src/cgame/cg_buildable.cpp [hunk 1]
```diff
@@ -708,16 +708,9 @@ void CG_InitBuildables()
...
 			{
-				if ( trap_FS_FOpenFile( filename, &f, fsMode_t::FS_READ ) > 0 )
-				{
-					//file exists so close it
-					trap_FS_FCloseFile( f );
-
-					cg_buildables[ buildable ].sounds[ j ].sound = trap_S_RegisterSound( filename, false );
-				}
-				else
+				if ( ( cg_buildables[ buildable ].sounds[ j ].sound = trap_S_RegisterSound( filename, false ) ) < 0 )
```
Comment (mbasaglia) [inline src/cgame/cg_buildable.cpp]: Maybe assignment and comparison in one go is a bit hard to read

**Comment by illwieckz** [comment]
the code does the job

File: src/cgame/cg_buildable.cpp (hunk 1 above)
Comment (illwieckz) [reply src/cgame/cg_buildable.cpp]: yolo

Comment (DolceTriade) [reply src/cgame/cg_buildable.cpp]: Done.

File: src/shared/bg_voice.cpp [hunk 2]
```diff
@@ -386,23 +386,20 @@ static voiceTrack_t *BG_VoiceParseCommand( int handle )
...
 		{
 			int  line;
 			char filename[ MAX_QPATH ];
 
 			trap_Parse_SourceFileAndLine( handle, filename, &line );
 			Log::Warn( "BG_VoiceParseCommand(): "
-			            "track \"%s\" referenced on line %d of %s does not exist",
-			            token.string, line, filename );
+						"track \"%s\" referenced on line %d of %s does not exist",
+						token.string, line, filename );
```
Comment (illwieckz) [inline src/shared/bg_voice.cpp]: I guess your IDE was a bit zealous on this spacing ? :grin: 

The spaces were not indenting spaces but aligning spaces (something an automatic styler can't tell), if we go the indentation way, I guess 1 tab is enough.

That was the obligatory bikeshedding comment. :bike:

**Comment by illwieckz** [comment]
Not perfect!

Running this code makes these warning occurring:

```
Warn: Sound file sound/buildables/arm/construct1 not found. 
Warn: Couldn't load sound sound/buildables/arm/construct1, it's empty! 
Warn: Sound file sound/buildables/drill/construct1 not found. 
Warn: Couldn't load sound sound/buildables/drill/construct1, it's empty! 
Warn: Sound file sound/buildables/spiker/pain1 not found. 
Warn: Couldn't load sound sound/buildables/spiker/pain1, it's empty! 
Warn: Sound file sound/buildables/spiker/pain2 not found. 
Warn: Couldn't load sound sound/buildables/spiker/pain2, it's empty! 
Warn: Sound file sound/buildables/medistat/construct1 not found. 
Warn: Couldn't load sound sound/buildables/medistat/construct1, it's empty! 
Warn: Sound file sound/buildables/mgturret/construct1 not found. 
Warn: Couldn't load sound sound/buildables/mgturret/construct1, it's empty! 
Warn: Sound file sound/buildables/hive/construct1 not found. 
Warn: Couldn't load sound sound/buildables/hive/construct1, it's empty! 
Warn: Sound file sound/buildables/spiker/construct1 not found. 
Warn: Couldn't load sound sound/buildables/spiker/construct1, it's empty! 
Warn: Sound file sound/buildables/rocketpod/construct1 not found. 
Warn: Couldn't load sound sound/buildables/rocketpod/construct1, it's empty! 
Warn: Sound file sound/buildables/reactor/construct1 not found. 
Warn: Couldn't load sound sound/buildables/reactor/construct1, it's empty! 
Warn: Sound file sound/buildables/repeater/construct1 not found. 
Warn: Couldn't load sound sound/buildables/repeater/construct1, it's empty! 
```

So let's take a look to the special case of the mgturret:

```
Warn: Sound file sound/buildables/mgturret/construct1 not found. 
Warn: Couldn't load sound sound/buildables/mgturret/construct1, it's empty! 
```

Do we have this sound in pk3? no:

```
unvanquished_0.25.0.pk3: sound/buildables/acid_tube/construct1.wav
unvanquished_0.25.0.pk3: sound/buildables/alien/construct1.wav
unvanquished_0.25.0.pk3: sound/buildables/barricade/construct1.wav
unvanquished_0.25.0.pk3: sound/buildables/human/construct1.wav
unvanquished_0.25.0.pk3: sound/buildables/overmind/construct1.wav
unvanquished_0.25.0.pk3: sound/buildables/telenode/construct1.wav
```

In fact, if `sound/buildables/mgturret/construct1` is empty, it must fallback on `sound/buildables/human/construct1`. That's how it's expected to work: use the default per-team sound if there is no per-building sound defined.

That's fallback mechanism is currently broken.

**Comment by DolceTriade** [comment]
Will address. THanks for testing.

**Comment by illwieckz** [comment]
Notice that my previous comment not only said it displayed some _warnings_, but also broke the fallback mechanism (just to be sure any fix not only hide the _warnings_).

File: src/cgame/cg_buildable.cpp [hunk 3]
```diff
@@ -708,7 +708,8 @@ void CG_InitBuildables()
 
 			if ( cg_buildables[ buildable ].sounds[ j ].enabled )
 			{
-				if ( ( cg_buildables[ buildable ].sounds[ j ].sound = trap_S_RegisterSound( filename, false ) ) < 0 )
+				cg_buildables[ buildable ].sounds[ j ].sound = trap_S_RegisterSound( filename, false )
```
Comment (illwieckz) [inline src/cgame/cg_buildable.cpp]: you forgot the `;`

File: src/cgame/cg_main.cpp [hunk 4]
```diff
@@ -1034,37 +1034,37 @@ static void CG_RegisterSounds()
...
 
-	cgs.media.lCannonWarningSound = trap_S_RegisterSound( "models/weapons/lcannon/warning.wav", false );
-	cgs.media.lCannonWarningSound2 = trap_S_RegisterSound( "models/weapons/lcannon/warning2.wav", false );
+	cgs.media.lCannonWarningSound = trap_S_RegisterSound( "models/weapons/lcannon/warning", false );
+	cgs.media.lCannonWarningSound2 = trap_S_RegisterSound( "models/weapons/lcannon/warning2", false );
 
-	cgs.media.rocketpodLockonSound = trap_S_RegisterSound( "sound/rocketpod/lockon.wav", false );
+	cgs.media.rocketpodLockonSound = trap_S_RegisterSound( "sound/rocketpod/lockon", false );
 
 	cgs.media.timerBeaconExpiredSound = trap_S_RegisterSound( "sound/feedback/beacon-timer-expired.ogg", false );
```
Comment (illwieckz) [inline src/cgame/cg_main.cpp]: note that you must now be able to remove the `.ogg` extension too

File: src/sgame/sg_spawn.cpp [hunk 5]
```diff
@@ -1015,7 +1015,7 @@ Every map should have exactly one.
 
 === KEYS ===
 ; message: Text to print during connection process. Used for the name of level.
-; music: path/name of looping .wav file used for level's music (eg. music/sonic5.wav).
+; music: path/name of looping  file used for level's music (eg. music/sonic5).
```
Comment (illwieckz) [inline src/sgame/sg_spawn.cpp]: > looping sound file

Comment (illwieckz) [reply src/sgame/sg_spawn.cpp]: there is a patch doing that in #1023

File: src/cgame/cg_main.cpp (hunk 4 above)
Comment (illwieckz) [reply src/cgame/cg_main.cpp]: there is a patch doing that in #1023

**Pull request merged**